src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -1669,6 +1669,7 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
               val tree1 = addBody(ddef, target)
               (new ChangeOwnerTraverser(target, tree1.symbol))(tree1.rhs)
               debuglog("changed owners, now: " + tree1)
+              if (target.hasFlag(Flags.SYNCHRONIZED)) ddef.symbol.setFlag(Flags.SYNCHRONIZED)
```

Comment:
I never looked at the implementation of `SpecializeTypes`, and I couldn't see the answer immediately, so let me just ask: why is the flat not copied when the `ddef.symbol` is created?